src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -855,8 +855,14 @@ public void waitNoPendingTasksOnAll() throws Exception {
             @Override
             public void run() {
                 for (Client client : clients()) {
+                    ClusterHealthResponse clusterHealth = client.admin().cluster().prepareHealth().setLocal(true).get();
+                    assertThat("client " + client  + " still has fetch shard started", clusterHealth.getNumberOfFetchShardStarted(), equalTo(0));
+                    assertThat("client " + client  + " still has fetch shard store", clusterHealth.getNumberOfFetchShardStore(), equalTo(0));
                     PendingClusterTasksResponse pendingTasks = client.admin().cluster().preparePendingClusterTasks().setLocal(true).get();
                     assertThat("client " + client + " still has pending tasks " + pendingTasks.prettyPrint(), pendingTasks, Matchers.emptyIterable());
+                    clusterHealth = client.admin().cluster().prepareHealth().setLocal(true).get();
```

Comment:
I might be missing something but why cannot we just use the number of pending tasks from the first health response and get rid of the pending tasks and the second health requests? If there is a reason for it, I think it might be worth adding it as a comment.